sample-app/src/components/FilterSearch.tsx

```diff
@@ -114,6 +115,10 @@ export default function FilterSearch ({
                 setInput(optionValue);
                 const result = filterSearchResponse?.sections[sectionIndex].results[optionIndex];
                 if (result?.filter) {
+                  if (selectedFilterOptionRef.current) {
+                    answersActions.setFilterOption({ ...selectedFilterOptionRef.current, selected: false });
```

Comment:
little out of scope but do we need the spread operator for the filter here and in setFilterOption?

Reply:
yea, setFilterOption expects type `SelectableFilter` (Filter's fields + selected field) so we would need to construct a new object with all the fields from the result's filter and the selected boolean.

Reply:
ah yeah I'm blind >.<